common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -868,8 +872,23 @@ class @Problem
       next_index = parseInt(hint_index) + 1
     $.postWithPrefix "#{@url}/hint_button", hint_index: next_index, input_id: @id, (response) =>
       if response.success
-        @render(response.html, @focus_on_hint_notification)
-        hint_container = @.$('.problem-hint')
+        hint_msg_container = @.$('.problem-hint .notification-message')
         hint_container.attr('hint_index', response.hint_index)
+        hint_msg_container.html(response.msg)
+        # Update any Mathjax entries
+        MathJax.Hub.Queue [
```

Comment:
I'm somewhat surprised that you can typeset before the notification is shown (visible). I tested a simple example and it seems to work, but I wonder if there is a race condition.